src/contrib/mlir/compiler/pass/mlir_subgraph_extraction.hpp

```diff
@@ -73,10 +71,13 @@ namespace ngraph
                 // Unique ID for this sub-graph.
                 int m_graph_id;
                 // Actual nodes of the sub-graph
-                NodeSet m_nodes;
-                // Predecessor to head nodes in the sub-graph.
-                NodeSet m_input_nodes;
-                NodeSet m_output_nodes;
+                NodeVector m_nodes;
+                // Predecessor to head nodes in the sub-graph. Both containers have the same
+                // elements. Set is only used for efficient look-up operations.
+                NodeVector m_input_node_vector;
```

Comment:
Why the order of input/output nodes causes non-determinism ? Rather avoid the duplication of data if not needed.

Reply:
IIRC, I saw function arguments in different order but I can double check if you think this is a problem. For outputs, I did it for consistency. I'm not sure if there is order is required for them. Having two container for performance reasons is usually acceptable (see VectorSet in LLVM). Do you have any other suggestion?